### ISSUE TITLE
Migrations fails to adds corresponding using when using enums

### ISSUE THREAD

**Issue by Bartmax** [aspnet/EntityFramework#6193]
Add a migration when a model has an enum fails to include the using for the appropiate namespace on the migration file and results in compilation error.

```
migrationBuilder.AddColumn<int>(
                name: "EnumName",
                table: "TableWithEnum",
                nullable: false,
                defaultValue: EnumName.None);
```

adding the using by hand solves the issue.

**Comment by rowanmiller** [comment]
Migrations actually shouldn't use the enum at all (since you may remove the enum from the model at some point in the future). It should use the underlying value (i.e. `0` in this case). I thought we had a dupe of this already, but I couldn't find it.

**Comment by rowanmiller** [comment]
@Bartmax which versions of the packages are you using? We thought this was fixed in RTM (but maybe we missed a case)

**Comment by Bartmax** [comment]
@rowanmiller I'm using 1.0 (RTM). And yup, It happened that I removed a type and I was unable to work with migrations, so using the underlying type makes lot of sense.

Just in case, the enum is defined as:
```
[Flags]
public enum MyEnum
{
    None = 0,
    First = 1,
    Second = 2,
    Third = 4,
    ...
}
```

**Comment by smitpatel** [comment]
#4508 was older issue. Looks like we have regressed.

**Comment by rowanmiller** [comment]
Thanks for the info - assigning to fix in 1.1

**Comment by bricelam** [comment]
I found the root cause here. Don't worry, @smitpatel, it's not a regression.

Because a non-nullable column is being added, the model differ is synthesizing a default. However, it is not unwrapping the enum type before getting a default value.
